test/tests/functional/pbs_multi_sched.py

```diff
@@ -284,135 +288,124 @@ def test_job_queue_partition(self):
         j = Job(TEST_USER1, attrs={ATTR_queue: 'wq1',
                                    'Resource_List.select': '1:ncpus=1'})
         jid = self.server.submit(j)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
-        nodes = ['vnode[0]']
-        self.check_vnodes(j, nodes, jid)
-        self.server.schedulers['sc1'].log_match(
-            str(jid) + ';Job run', max_attempts=10,
+        self.check_vnodes(j, ['vnode[0]'], jid)
+        self.scheds['sc1'].log_match(
+            jid + ';Job run', max_attempts=10,
             starttime=self.server.ctime)
         p1 = {'partition': 'P1'}
         self.server.manager(MGR_CMD_SET, QUEUE, p1, id='wq4')
         j = Job(TEST_USER1, attrs={ATTR_queue: 'wq4',
                                    'Resource_List.select': '1:ncpus=1'})
         jid = self.server.submit(j)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid)
-        nodes = ['vnode[0]']
-        self.check_vnodes(j, nodes, jid)
-        self.server.schedulers['sc1'].log_match(
-            str(jid) + ';Job run', max_attempts=10,
+        self.check_vnodes(j, ['vnode[0]'], jid)
+        self.scheds['sc1'].log_match(
+            jid + ';Job run', max_attempts=10,
             starttime=self.server.ctime)
 
     def test_preemption_highp_queue(self):
         """
         Test preemption occures only within queues
-        which are assigned to same scheduler
+        which are assigned to same partition
         """
-        prio = {'Priority': 150}
-        self.server.manager(MGR_CMD_SET, QUEUE, prio, id='wq3')
+        self.common_setup()
+        prio = {'Priority': 150, 'partition': 'P1'}
         self.server.manager(MGR_CMD_SET, QUEUE, prio, id='wq4')
         j = Job(TEST_USER1, attrs={ATTR_queue: 'wq1',
                                    'Resource_List.select': '1:ncpus=2'})
         jid1 = self.server.submit(j)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid1)
-        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
-                                   'Resource_List.select': '1:ncpus=2'})
-        jid2 = self.server.submit(j)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid2)
-        j = Job(TEST_USER1, attrs={ATTR_queue: 'wq3',
-                                   'Resource_List.select': '1:ncpus=2'})
-        jid3 = self.server.submit(j)
-        self.server.expect(JOB, {'job_state': 'Q'}, id=jid3)
+        t = int(time.time())
         j = Job(TEST_USER1, attrs={ATTR_queue: 'wq4',
                                    'Resource_List.select': '1:ncpus=2'})
         jid4 = self.server.submit(j)
         self.server.expect(JOB, {'job_state': 'R'}, id=jid4)
-        t = int(time.time())
         j = Job(TEST_USER1, attrs={ATTR_queue: 'wq4',
                                    'Resource_List.select': '1:ncpus=2'})
         jid5 = self.server.submit(j)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid5)
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid5)
```

Comment:
Check to make sure the comment is set prior to checking for job_state Q.  It is possible that your expect() happens before the scheduler sees the job.

Reply:
Done